### PR TITLE
Use GitHub Actions

### PR DESCRIPTION
**Pull request by aminya**
### Description of the change

Use GitHub Actions instead of Travis and Appveyor. 

### Benefits
- GitHub Actions is much faster than the other CI services and allows parallel runs 
- GitHub Actions it is much easier to configure. 
- Atom is a GitHub product, so we should use GitHub Actions.
- Travis is making its services limited and monetary.   
https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing

### Verification 
The CI passes: https://github.com/aminya/node-pathwatcher/actions/runs/359170418

### Drawbacks
none

### Release Notes
N/A

Closes #93

### CODE REVIEW

**Comment by aminya** [comment]
@darangi @sadick254

 I plan to change all the CI of the packages to GitHub Actions. Atom is a GitHub product, so we should use GitHub Actions. The configuration is the same for all packages, so no review is needed for each individual package as long as the tests pass.

**Comment by sadick254** [comment]
Hey @aminya that would be awesome. Migrating to Github Actions was something on our plan. I encourage you to go ahead with the change although we might not be able to look at the PRs at the moment.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+jobs:
+  Test:
+    if: "!contains(github.event.head_commit.message, '[skip ci]')"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          # - macos-latest
```
Comment (sadick254) [inline .github/workflows/ci.yml:15]:
```suggestion
          - macos-latest
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          - windows-latest
+        node_version:
+          - 10
```
Comment (sadick254) [inline .github/workflows/ci.yml:18]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          - 12
+          # - 14
```
Comment (sadick254) [inline .github/workflows/ci.yml:20]:
```suggestion
          - 14
```

File: appveyor.yml [hunk 4]
```diff
@@ -1,26 +1,6 @@
...
-  # AppVeyor still does not support Node v12: https://github.com/appveyor/ci/issues/2921
-  - ps: Update-NodeJsInstallation (Get-NodeJsLatestBuild $env:nodejs_version) $env:PLATFORM
-  - npm install
-
-test_script:
-  - node -e "console.log(`${process.version} ${process.arch} ${process.platform}`)"
-  - npm --version
-  - npm test
-
+# dummy appveyor
```
Comment (sadick254) [inline appveyor.yml:1]: Can we delete this file as well.

**Pull request merged**